src/translator/batch_translator.h

```diff
@@ -49,6 +50,9 @@ class BatchTranslator {
   std::vector<Ptr<Scorer>> scorers_;
   Ptr<data::ShortlistGenerator const> slgen_;
   const void * model_memory_;
+  const size_t model_memory_size_;
+  const void * shortlist_memory_;
+  const size_t shortlist_memory_size_;
```

Comment:
naming with camelcase and not snakecase (marian does this?). Also use the struct or a container of struct to bundle all.